src/signalrclient/connection_impl.cpp

```diff
@@ -375,20 +382,22 @@ namespace signalr
                 }
             });
 
-        std::thread([disconnect_cts, connect_request_done, connect_request_lock, callback, weak_connection]()
-        {
-            disconnect_cts->wait(5000);
-
+        timer(m_scheduler, [disconnect_cts, connect_request_done, connect_request_lock, callback](std::chrono::milliseconds duration) {
             bool run_callback = false;
             {
                 std::lock_guard<std::mutex> lock(*connect_request_lock);
+
                 // no op after connection started successfully
                 if (*connect_request_done == false)
                 {
+                    if (!disconnect_cts->is_canceled() && duration < std::chrono::seconds(5))
```

Comment:
Should we make this 5 second limit configurable eventually?

Reply:
Eventually